spec/predicates/runtime-trace.md

```diff
@@ -80,49 +105,68 @@ Identifies the specific monitor instance used to trace the runtime.
 
 URI indicating the monitor’s identity.
 
-`build` _object_, _required_
+`process` _object_, _required_
 
-Feedback required: Should this predicate be scoped to runtime traces of builds only or generalize to runtime traces of any activity?
+`process.HostID` _string (URI)_, _required_
 
-`builder.builderId` _string (URI)_, _required_
+URI indicating the process host’s identity. Ex: GitLab Runner ID
 
-URI indicating the builder’s identity. Ex: GitLab Runner ID
+`process.type` _string (TypeURI)_, _required_
 
-`build.type` _string (TypeURI)_, _required_
+URI indicating the type of process performed.
 
-URI indicating the type of build performed.
-
-`build.event` _string_, _required_
+`process.event` _string_, _required_
 
 String identifying the specific job or task associated with the attestation.
 
 `invocation` _object_, _optional_
 
-Identifies the event that initiated the monitor. Similar to `invocation` in SLSA Provenance.
+Identifies the event that initiated the monitor.
 
 `invocation.configSource` _object_, _optional_
 
 Effectively a pointer to the monitor's configuration.
 
 `invocation.tracePolicy` _object_, _optional_
 
-Feedback required: Should tracingPolicy point outwards to a separate attestation with the same subject that wraps around the actual policy? The policy can be more securely defined and distributed (signed for by developer keys rather than online keys for example).
+Feedback required: Should tracingPolicy point outwards to a separate attestation
+with the same subject that wraps around the actual policy? The policy can be
+more securely defined and distributed (signed for by developer keys rather than
+online keys for example).
```

Comment:
It strikes me that the `tracePolicy` wouldn't actually change as often as the runtime traces are generated (correct me if I'm wrong), so I think it would definitely make sense to include a reference to a separate policy attestation that can be reused for any trace it applies to.

Reply:
I think I was in favour of having it be a separate document but I think either @pxp928 or @nadgowdas noted that we need to unambiguously map the trace policy to the trace so that we can reason about it when writing policies for `monitorLog`. For a separate document, I guess the monitor would have to understand that a configured trace policy matches the corresponding attestation? Perhaps a TODO past the v0.1 spec?

Reply:
Using a separate document for the policy is a better solution in retrospect. I agree that it would not change that often but it would have to be published alongside the runtime attestation each time to ensure which policy was set when the trace was captured. If the policy for example is set to not capture any network logs, we cannot attest that during the build time, the pipeline did not reach out to the internet.